extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -108,16 +109,129 @@ class MultipleBackendTask extends ReplicateObject {
         }, cb);
     }
 
-    _getAndPutMPUPartOnce(sourceEntry, destEntry, part, uploadId, log, done) {
-        log.debug('getting object part', { entry: sourceEntry.getLogInfo() });
+    /**
+     * This is a retry wrapper for calling _getRangeAndPutMPUPartOnce.
+     * @param {ObjectQueueEntry} sourceEntry - The source object entry
+     * @param {ObjectQueueEntry} destEntry - The destination object entry
+     * @param {String} range - The range to get an object with
+     * @param {Number} partNumber - The part number for the current part
+     * @param {String} uploadId - The upload ID of the initiated MPU
+     * @param {Werelogs} log - The logger instance
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    _getRangeAndPutMPUPart(sourceEntry, destEntry, range, partNumber, uploadId,
+        log, cb) {
+        this.retry({
+            actionDesc: 'stream part data',
+            logFields: { entry: sourceEntry.getLogInfo() },
+            actionFunc: done => this._getRangeAndPutMPUPartOnce(sourceEntry,
+                destEntry, range, partNumber, uploadId, log, done),
+            shouldRetryFunc: err => err.retryable,
+            log,
+        }, cb);
+    }
+
+    /**
+     * Calculate the byte size of the given range.
+     * @param {String} range - The range request string (e.g., 'bytes=0-1').
+     * @return {Number} The number of bytes that the range represents.
+     */
+    _getRangeSize(range) {
```

Comment:
Instead of manipulating HTTP range strings, it could be simpler to pass around range JS objects (eg. with `start` and `end` attributes) to avoid any unnecessary parsing, only when preparing the HTTP GET request will we create the range header. Not talking about performance here but rather extra code complexity.